test/xbgpu/test_engine.py

```diff
@@ -1289,42 +1379,65 @@ def heap_factory(batch_index: int, present: np.ndarray) -> list[spead2.send.Heap
             assert stream_diff.diff("output_x_clipped_visibilities_total") == n_vis
             assert xbengine.sensors[f"{corrprod_output.name}.xeng-clip-cnt"].value == n_vis
 
-    def _patch_get_in_item(
+    def _patch_method(
         self,
         monkeypatch: pytest.MonkeyPatch,
+        method_name: str,
+        class_type: type[_T],
         count: int,
         client: aiokatcp.Client,
         requests: Iterable[Iterable],
     ) -> list[int]:
-        """Patch :meth:`~.BPipeline._get_in_item` to make requests partway through the stream.
+        """Patch `method_name` of class `class_type` to make requests during operation.
 
         The returned list will be populated with the value of the
         ``steady-state-timestamp`` sensor immediately after executing the
-        request.
+        requests.
 
         Parameters
         ----------
+        method_name
+            The method intended to be patched.
+        class_type
+            The class which has an attribute `method_name`.
         count
-            Counting from 1, the index of the call to `_get_in_item` to trigger
+            Counting from 1, the index of the call to `method_name` to trigger
             `requests`. `requests` are made prior to the n-th call to
             `_get_in_item`.
         requests
             A list of tuples in the format ("request-name", arg1, arg2, ...).
+
+        Returns
+        -------
+        timestamp
+            The steady-state timestamp obtained after executing `requests`.
+
+        Raises
+        ------
+        AttributeError
+            If `class_type` does not have a `method_name`.
+
+        .. todo::
+
+            NGC-1568: Update the count logic to count from zero instead of one.
         """
         counter = 0
         timestamp = []
-        orig_get_in_item = BPipeline._get_in_item
+        try:
+            orig_method = getattr(class_type, method_name)
+        except AttributeError:
+            raise AttributeError(f"Class {class_type} does not have method {method_name}")
```

Comment:
Is it really necessary to catch an AttributeError only to raise another AttributeError?

If for some reason it is (e.g. if the original exception message is unhelpful), then append `from None` here so that you don't get the message that an exception was raised during exception handling along with both exception tracebacks.